front/app/component-library/components/CollapsibleContainer/index.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { ReactNode, useEffect, useState } from 'react';
+
+import { CSSTransition } from 'react-transition-group';
+import styled from 'styled-components';
+
+import useInstanceId from '../../hooks/useInstanceId';
+import { colors, isRtl } from '../../utils/styleUtils';
+import Box, { BoxMarginProps, BoxPaddingProps, BoxWidthProps } from '../Box';
+import Icon from '../Icon';
+import Title from '../Title';
+
+type CollapsibleContainerProps = {
+  title: string;
+  children: ReactNode;
+  isOpenByDefault?: boolean;
+} & BoxMarginProps &
+  BoxWidthProps &
+  BoxPaddingProps;
+
+const ChevronIcon = styled(Icon)`
+  fill: ${colors.black};
+  margin-left: 20px;
+  transition: fill 80ms ease-out, transform 200ms ease-out;
+`;
+
+const TitleButton = styled(Box)`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  cursor: pointer;
+  width: 100%;
+  text-align: left;
+  ${isRtl`
+    text-align: right;
+    direction: rtl;
+    ${ChevronIcon} {
+        transform: rotate(180deg);
+    }
+`}
+  &.expanded {
+    ${ChevronIcon} {
+      transform: rotate(90deg);
+    }
+  }
+  &:hover {
+    ${ChevronIcon} {
+      fill: #000;
+    }
+  }
+`;
+
+const CollapseContainer = styled(Box)`
+  margin-bottom: 16px;
+  opacity: 1;
+  display: flex;
+  flex-wrap: wrap;
+  transition: 'all 1000ms cubic-bezier(0.165, 0.84, 0.44, 1)';
+  will-change: opacity, height;
+
+  ${isRtl`
+    text-align: right;
+    direction: rtl;
+`}
+
+  &.expanded-enter {
+    opacity: 0;
+    max-height: 0px;
+    overflow: hidden;
+
+    &.expanded-enter-active {
+      opacity: 1;
+      max-height: 1000px;
+      overflow: hidden;
+    }
+  }
+
+  &.expanded-enter-done {
+    opacity: 1;
+    overflow: visible;
+  }
+
+  &.expanded-exit {
+    opacity: 1;
+    max-height: 1000px;
+    overflow: hidden;
+
+    &.collapsed-exit-active {
+      opacity: 0;
+      max-height: 1000px;
+      overflow: hidden;
+    }
+  }
+`;
+
+const CollapsibleContainer = ({
+  isOpenByDefault,
+  title,
+  children,
+}: CollapsibleContainerProps) => {
+  const [isExpanded, setIsExpanded] = useState(isOpenByDefault);
+  const uuid = useInstanceId();
+
+  useEffect(() => {
+    setIsExpanded(isOpenByDefault);
+  }, [isOpenByDefault]);
+
+  const handleChange = () => {
+    setIsExpanded(!isExpanded);
+  };
+
+  return (
+    <Box
+      display="flex"
+      flexDirection="column"
+      background={colors.white}
+      px="16px"
+      borderRadius="4px"
+      minWidth="300px"
```

Comment:
I would keep styles out of this component to make it more broadly reusable. The component should handle the functional stuff (open/closed, a11y) but not the styles. We _could_ then use this to build reusable UI elements if we want to reuse styles across the app.

Reply:
> but not the styles
Hmm I'm not sure if I agree that styles shouldn't be included - isn't the purpose of a component library to have styled components that can be re-used across the platform?

I agree that some things (E.g. max-width) can be removed for more re-usability though :+1:

Reply:
Component library components should be style-free and then components with certain styles can be created in the UI folder imo.

Reply:
@brentguf Our component library today is a collection of pre-styled/designed UI elements already though right? :thinking: (E.g. Button, Dropdown, Search Input are all components with specific styling set on them).

Reading through [this article](https://www.uxpin.com/studio/blog/ui-component-library/), they suggest that the point of a component library is to have components with certain styles set, so designers & devs can re-use them across the platform.

You're suggesting though that a component library should be purely functional with no styles? I'm just a bit confused, since it doesn't feel to me like that's how our library is set up today though - just want to understand a bit better :slightly_smiling_face: